visionOS Build Failure fix.

robert-jenner opened the pull request:
#### d8737afd6ebe87583a510675a5f60a212352e28b
<pre>
visionOS Build Failure fix.
<a href="https://rdar.apple.com/163511314">rdar://163511314</a>

Reviewed by Aditya Keerthi.

Temporarily ignore deprecations in: Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm.

* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
(-[WKFullScreenParentWindowState initWithWindow:]):
(-[WKFullScreenWindowController didEnterVideoFullscreen]):
(-[WKFullScreenWindowController didExitVideoFullscreen]):
(-[WKFullScreenWindowController _performSpatialFullScreenTransition:completionHandler:]):
(-[WKFullScreenWindowController toggleSceneDimming]):

Canonical link: <a href="https://commits.webkit.org/302199@main">https://commits.webkit.org/302199@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/411389ff046ad786fa8f1af9f1f5fb8e8d8abf0f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/128347 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/623 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/39179 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/135739 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/79814 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 ios-apple 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/559 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/156/builds/495 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/97695 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/79814 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/131295 "Passed tests") | [⏳ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/iOS-26-Simulator-WK2-Tests-EWS "Waiting to run tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/114980 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/78289 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | ⏳ 🛠 vision-apple 
| | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-26-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/79026 "Built successfully") | | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/33572 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/138193 "Failed to checkout and rebase branch from PR 53044") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/468 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/156/builds/495 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/138193 "Failed to checkout and rebase branch from PR 53044") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/505 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/111322 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/138193 "Failed to checkout and rebase branch from PR 53044") | | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/164/builds/386 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/29870 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/52785 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/20043 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/519 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/63694 "Built successfully") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/157/builds/415 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/163/builds/481 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/480 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
```diff
@@ -2008,12 +2016,14 @@ - (void)_performSpatialFullScreenTransition:(BOOL)enter completionHandler:(Compl
+ALLOW_DEPRECATED_DECLARATIONS_END
         } completion:nil];
     }
 
```
I'd move the `ALLOW_DEPRECATED_DECLARATIONS_END` down here, right after the if. In the middle of the block is strange.

robert-jenner replied:
Understood. I think I addressed this.

pxlcoder commented on Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
```diff
@@ -594,7 +594,9 @@ @interface WKFullScreenParentWindowState : NSObject
 @property (nonatomic, readonly) MRUISceneResizingBehavior sceneResizingBehavior;
+ALLOW_DEPRECATED_DECLARATIONS_BEGIN
 @property (nonatomic, readonly) MRUIDarknessPreference preferredDarkness;
+ALLOW_DEPRECATED_DECLARATIONS_END
```
Is the change here really necessary?

robert-jenner replied:
Probably not actually. Removed.

pxlcoder commented on Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
```diff
@@ -2008,13 +2014,15 @@ - (void)_performSpatialFullScreenTransition:(BOOL)enter completionHandler:(Compl
         [UIView animateWithDuration:kDarknessAnimationDuration animations:^{
             stage.preferredDarkness = targetDarkness;
         } completion:nil];
+ALLOW_DEPRECATED_DECLARATIONS_END
```
I'd move this down after the brace below ("right after the if")

robert-jenner replied:
Oops. Fixed now.

robert-jenner committed:
visionOS Build Failure fix.
rdar://163511314

Reviewed by Aditya Keerthi.

Temporarily ignore deprecations in: Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm.

* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
(-[WKFullScreenParentWindowState initWithWindow:]):
(-[WKFullScreenWindowController didEnterVideoFullscreen]):
(-[WKFullScreenWindowController didExitVideoFullscreen]):
(-[WKFullScreenWindowController _performSpatialFullScreenTransition:completionHandler:]):
(-[WKFullScreenWindowController toggleSceneDimming]):

Canonical link: https://commits.webkit.org/302199@main
